### PR TITLE
Sre not gc dependent

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/reflection.c [hunk 1]
```diff
@@ -189,34 +186,28 @@ clear_cached_object (MonoDomain *domain, gpointer o, MonoClass *klass)
...
+	free_reflected_entry (key);
 }
-#endif
 
 void
 mono_reflection_cleanup_domain (MonoDomain *domain)
 {
 	if (domain->refobject_hash) {
-/*let's avoid scanning the whole hashtable if not needed*/
-#ifdef REFENTRY_REQUIRES_CLEANUP
```
Comment (lambdageek) [inline mono/metadata/reflection.c]: @joncham Is this bit okay?

**Comment by vargaz** [comment]
build

File: mono/metadata/reflection-cache.h [hunk 2]
```diff
@@ -30,16 +30,21 @@ reflected_equal (gconstpointer a, gconstpointer b);
...
-#define FREE_REFENTRY(entry)
-#endif
+static inline ReflectedEntry*
+alloc_reflected_entry (MonoDomain *domain)
+{
+	if (!mono_gc_is_moving ())
+		return g_new0 (ReflectedEntry, 1);
+	else
+		return (ReflectedEntry *)mono_mempool_alloc (domain->mp, sizeof (ReflectedEntry));
+}
```
Comment (vargaz) [inline mono/metadata/reflection-cache.h]: This is what the old code did, but I have no idea why. Its probably a leftover.

File: mono/metadata/reflection.c (hunk 1 above)
Comment (joncham) [inline mono/metadata/reflection.c]: Seems okay

File: mono/metadata/reflection-cache.h (hunk 2 above)
Comment (joncham) [inline mono/metadata/reflection-cache.h]: I agree mempool should work on Boehm as well.

Comment (vargaz) [inline mono/metadata/reflection-cache.h]: Changed this to use malloc, so we can free the memory for dynamic methods, i.e. from mono_method_clear_object ().

**Pull request merged**